File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/GeneratorConfig.java
package io.github.potjerodekool.openapi.common;

import io.github.potjerodekool.codegen.Language;

import java.util.Map;

public record GeneratorConfig(Language language,
                              String basePackageName,
                              Map<String, Boolean> features) {

    public String configPackageName() {
        return basePackageName + ".config";
    }

    public boolean isFeatureEnabled(final String feature) {
        return Boolean.TRUE.equals(features.get(feature));
    }

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/GeneratorConfig.kt
package io.github.potjerodekool.openapi.common

import io.github.potjerodekool.codegen.Language

data class GeneratorConfig(
    val language: Language,
    val basePackageName: String,
    val features: Map<String, Boolean>
) {
    fun configPackageName(): String {
        return "$basePackageName.config"
    }

    fun isFeatureEnabled(feature: String): Boolean {
        return java.lang.Boolean.TRUE == features[feature]
    }
}
